### PR TITLE
First wrapper

### PR DESCRIPTION
**Pull request by marrabld**
All of the unit tests work on my machine.  @RobbieFernandez Would you mind sanity checking you can also run the tests.

I am wondering if returning the ctypes structures are not pythonic.  I wonder if I should convert them all to say namedtuples before returning.  thoughts ?   I am quite lazy though.  Is there away that we can change `__repr__` so that it looks like a tuple ? 

Need to make it pip installable too and updated documentation and add a how to.

### CODE REVIEW

File: src/emtmlibpy/emtmlibpy.py [hunk 1]
```diff
@@ -0,0 +1,795 @@
...
+    """
+
+    pminor = ctypes.POINTER(ctypes.c_int)
+    pmajor = ctypes.POINTER(ctypes.c_int)
+    libc.EMTMVersion.argtypes = [pminor, pmajor]
+
+    minor = ctypes.c_int(0)
+    major = ctypes.c_int(0)
+
+    libc.EMTMVersion(major, minor)
```
Comment (RobbieFernandez) [inline src/emtmlibpy/emtmlibpy.py:339]: Should you not be passing the pointers in here?

Comment (RobbieFernandez) [reply src/emtmlibpy/emtmlibpy.py:339]: Feel like this always returns (0, 0)

File: src/emtmlibpy/emtmlibpy.py [hunk 2]
```diff
@@ -0,0 +1,795 @@
...
+    libc.EMMeasurementCountFGS(bytes(family, 'UTF-8'),
+                               bytes(genus, 'UTF-8'),
+                               bytes(species, 'UTF-8'),
+                               ctypes.byref(n_point),
+                               ctypes.byref(n_box),
+                               ctypes.byref(n_3D_point),
+                               ctypes.byref(n_length),
+                               ctypes.byref(n_cpd_length))
+
+    FGS = namedtuple('FGS', 'point box xyz_point, length cpd_length')
```
Comment (RobbieFernandez) [inline src/emtmlibpy/emtmlibpy.py:542]: i think 
```python
'point box xyz_point, length cpd_length'
```
is meant to be 
```python
['point', 'box', 'xyz_point', 'length', 'cpd_length']
```

File: src/emtmlibpy/emtmlibpy.py [hunk 3]
```diff
@@ -0,0 +1,795 @@
...
+
+    It is sufficient to call this function once before making multiple calls to
+    EMGetPoint.
+    :return:
+    """
+
+    pn_bbox = ctypes.c_int(0)
+    r = libc.EMPointCount(ctypes.byref(pn_bbox))
+
+    PointCount = namedtuple('PointCount', 'total bbox')
```
Comment (RobbieFernandez) [inline src/emtmlibpy/emtmlibpy.py:573]: Same here, should be `['total', 'bbox']`

File: src/emtmlibpy/emtmlibpy.py [hunk 4]
```diff
@@ -0,0 +1,795 @@
...
+    length measurements (that is, the return value is the total number of
+    length and compound length measurements). Use pnCompound to
+    find the number of compound lengths
+    """
+
+    pn_compound = ctypes.c_int(0)
+
+    r = libc.EMLengthCount(pn_compound)
+
+    LengthCount = namedtuple('LengthCount', 'total compound')
```
Comment (RobbieFernandez) [inline src/emtmlibpy/emtmlibpy.py:682]: `['total', 'compound']`

**Review by RobbieFernandez — CHANGES_REQUESTED** [review]
Unit tests pass for me too :tada: 

Left a few comments with some changes :smile: 

As for returning the Ctypes struct, I don't see anything wrong with that. You can access the properties by name (as far as I can tell) so I'd call it Pythonic since it's like a duck-typed named tuple :snake:

**Comment by RobbieFernandez** [comment]
Disregard my named tuple feedback, I didn't know you could pass space-separated field names like that :open_mouth:

**Comment by marrabld** [comment]
Thanks for the suggested changes.  Ill get on to that very soon.

File: src/emtmlibpy/emtmlibpy.py (hunk 1 above)
Comment (RobbieFernandez) [reply src/emtmlibpy/emtmlibpy.py:339]: interesting! maybe they get passed by reference automatically? I'm not familiar with ctypes so possible I've misunderstood how this works

File: src/emtmlibpy/emtmlibpy.py [hunk 5]
```diff
@@ -0,0 +1,795 @@
+"""
+Abstraction library for libEMTLib.so from SeaGIS.
+Requires a licence to use, this is just a python wrapping function
+"""
+import ctypes
+from enum import IntEnum, auto
+from collections import namedtuple
+
+EMTM_MAX_CHARS = 1024
+libc = ctypes.CDLL('./emtmlibpy/libEMTMLib.so')
```
[the remark this answers is not in the document]
Comment (marrabld) [reply src/emtmlibpy/emtmlibpy.py:10]: Done.  Made this change.

File: src/emtmlibpy/emtmlibpy.py [hunk 6]
```diff
@@ -0,0 +1,795 @@
...
+        ('str_code', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_number', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_stage', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_activity', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_comment', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_9', ctypes.c_char * EMTM_MAX_CHARS),
+        ('str_att_10', ctypes.c_char * EMTM_MAX_CHARS)
+    ]
+
+    def __int__(self, str_op_code,
```
[the remark this answers is not in the document]
Comment (marrabld) [reply src/emtmlibpy/emtmlibpy.py:52]: uggghhghghg.  That means non of the values were initialised correctly either .  Fixed

File: src/emtmlibpy/emtmlibpy.py (hunk 1 above)
Comment (marrabld) [reply src/emtmlibpy/emtmlibpy.py:339]: Explicitly passing by ref now.

**Review by RobbieFernandez — APPROVED** [review]

**Pull request merged**